From 3bd323cefe18d80b275f64652ac531a031936822 Mon Sep 17 00:00:00 2001
From: Beleswar Padhi <b-padhi@ti.com>
Date: Wed, 12 Feb 2025 22:57:22 +0530
Subject: [PATCH 63/68] FROMLIST: remoteproc: k3-m4: Update rproc pointer
 within k3_rproc struct

The k3_m4_rproc struct originally did not have a (struct rproc *)
member. But with the data structures combined between DSP and M4
drivers, the k3_rproc struct expects the (struct rproc *) member to be
populated which would be used later in refactored common functions like
k3_rproc_mbox_callback().

Update the (struct rproc *) pointer in the probe routine.

Fixes: 26311608f97a ("FROMLIST: remoteproc: k3: Refactor mailbox rx_callback functions into common driver")
Signed-off-by: Beleswar Padhi <b-padhi@ti.com>
Tested-by: Judith Mendez <jm@ti.com>
Reviewed-by: Andrew Davis <afd@ti.com>
---
 drivers/remoteproc/ti_k3_m4_remoteproc.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/remoteproc/ti_k3_m4_remoteproc.c b/drivers/remoteproc/ti_k3_m4_remoteproc.c
index 291560fa899b..bfb28ee68231 100644
--- a/drivers/remoteproc/ti_k3_m4_remoteproc.c
+++ b/drivers/remoteproc/ti_k3_m4_remoteproc.c
@@ -62,6 +62,7 @@ static int k3_m4_rproc_probe(struct platform_device *pdev)
 	kproc = rproc->priv;
 	kproc->dev = dev;
 	kproc->data = data;
+	kproc->rproc = rproc;
 	platform_set_drvdata(pdev, rproc);
 
 	kproc->ti_sci = devm_ti_sci_get_by_phandle(dev, "ti,sci");
-- 
2.47.2

